map_test.go

```diff
@@ -1910,38 +1998,73 @@ func BenchmarkMarshaling(b *testing.B) {
 }
 
 func BenchmarkPerCPUMarshalling(b *testing.B) {
+	batch := 1024
 	newMap := func(valueSize uint32) *Map {
 		m, err := NewMap(&MapSpec{
 			Type:       PerCPUHash,
 			KeySize:    8,
 			ValueSize:  valueSize,
-			MaxEntries: 1,
+			MaxEntries: uint32(batch),
 		})
 		if err != nil {
 			b.Fatal(err)
 		}
 		return m
 	}
 
-	key := uint64(1)
-	val := []uint64{1, 2, 3, 4, 5, 6, 7, 8}
+	possibleCPUs, err := PossibleCPU()
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	val := make([]uint64, possibleCPUs)
+	for i := range val {
+		val[i] = uint64(i + 1)
+	}
 
 	m := newMap(8)
-	if err := m.Put(key, val[0:]); err != nil {
-		b.Fatal(err)
+	for i := 0; i < batch; i++ {
+		key := uint64(i + 1)
+		if err := m.Put(key, val); err != nil {
+			b.Fatal(err)
+		}
 	}
 	b.Cleanup(func() { m.Close() })
 
-	b.Run("reflection", func(b *testing.B) {
+	b.Run("Lookup", func(b *testing.B) {
 		b.ReportAllocs()
 		b.ResetTimer()
 
 		var value []uint64
 
 		for i := 0; i < b.N; i++ {
-			err := m.Lookup(unsafe.Pointer(&key), &value)
-			if err != nil {
-				b.Fatal("Can't get key:", err)
+			for j := 0; j < batch; j++ {
+				key := uint64(j + 1)
+				err := m.Lookup(unsafe.Pointer(&key), &value)
+				if err != nil {
+					b.Fatal("Can't get key:", err)
+				}
+			}
+		}
+	})
+	b.Run("BatchLookup", func(b *testing.B) {
+		b.ReportAllocs()
+		b.ResetTimer()
```

Comment:
Should this happen after allocating the buffers below?

Reply:
Same for
```
		var value []uint64
```
in the previous, but generally should be insignificant compared to `b.N`